projects/swimlane/ngx-charts/src/lib/common/base-chart.component.ts

```diff
@@ -209,4 +210,48 @@ export class BaseChartComponent implements OnChanges, AfterViewInit, OnDestroy,
 
     return results;
   }
+
+  toDataURL<T extends 'png' | 'jpg' | 'svg'>(
+    options: {
+      type?: T;
+      canvasOptions?: { pixelRatio?: number; transparentBackground?: boolean };
+    } = {}
+  ): Promise<string> {
+    if (this.getContainerDims() === null)
+      // If not browser
+      return null;
+
+    const chartEl = this.chartElement.nativeElement.firstElementChild;
+
+    const ops = {
+      width: this.width,
+      height: this.height,
+      ...(options.canvasOptions || {})
+    };
+    const { type } = options;
+    if (type === 'svg') {
+      return toSvg(chartEl, ops);
+    }
+    if (type === 'png') {
+      return toPng(chartEl, ops);
+    }
+    if (type === 'jpg') {
+      return toJpeg(chartEl, ops);
+    }
+
+    throw new Error(`Unable to convert to type ${type}`);
+  }
+
+  toCanvas(options: { pixelRatio?: number; transparentBackground?: boolean }): Promise<HTMLCanvasElement> {
+    if (this.getContainerDims() === null)
+      // If not browser
+      return null;
```

Comment:
Include brackets to match coding style.
```suggestion
    if (this.getContainerDims() === null) {
      // If not browser
      return null;
    }
```